code/game/objects/effects/mines.dm

```diff
@@ -57,6 +57,10 @@
 	var/radiation_amount
 
 /obj/effect/mine/dnascramble/mineEffect(mob/living/victim)
+	if(ishuman(victim))
+		var/mob/living/carbon/human/V = victim
+		if(V.species.flags & (RADIMMUNE | NO_DNA))

```

Comment:
You want radiation effects to apply to rad immune species in *most* cases; you just don't want those who lack DNA to get mutated.